### PR TITLE
Trace Log Upload Timing in BlobStoreRepository

### PR DESCRIPTION
**Pull request by original-brownbear**
We want more information on the write latencies. Lets trace log
the timing on uploading data blobs which involve almost no CPU during
uploading to gather some data.

### CODE REVIEW

**Review by idegtiarenko — APPROVED** [review]

File: server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java [hunk 1]
```diff
@@ -3475,8 +3475,17 @@ private void checkAborted() {
...
+                logger.trace("[{}] Writing [{}] to [{}]", metadata.name(), partName, shardContainer.path());
+                final long startMS = threadPool.relativeTimeInMillis();
                 shardContainer.writeBlob(partName, inputStream, partBytes, false);
+                logger.trace(
+                    "[{}] Writing [{}] of size [{}b] to [{}] took [{}ms]",
+                    metadata.name(),
+                    partName,
+                    partBytes,
+                    shardContainer.path(),
+                    threadPool.relativeTimeInMillis() - startMS
```
Comment (henningandersen) [inline server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java:3487]: a true nit, but I prefer to capture the `threadPool.relativeTimeInMillis()` explicitly right after the write to not have to worry about any time spent in any of the (obviously fast) method calls here.

**Review by henningandersen — APPROVED** [review]
LGTM.

File: server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java (hunk 1 above)
Comment (original-brownbear) [reply server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java:3487]: :) I think here you don't have any guarantees on the order of invocations anyway after JIT (which will inline the method calls anyway) so I'll leave it as is to save a line.

**Comment by original-brownbear** [comment]
Thanks both!

**Pull request merged**